concent_api/core/transfer_operations.py

```diff
@@ -107,95 +107,92 @@ def store_pending_message(
+        source_package_hash=report_computed_task.task_to_compute.package_hash,
+        result_package_path=get_storage_result_file_path(task_id, subtask_id),
+        result_size=report_computed_task.size,
+        result_package_hash=report_computed_task.package_hash,
+        authorized_client_public_key=authorized_client_public_key,
+        operation=operation,
+        token_expiration_deadline=calculate_token_expiration_deadline(operation, report_computed_task)
     )
 
 
 def _create_file_transfer_token(
-    report_computed_task: message.tasks.ReportComputedTask,
-    authorized_client_public_key: bytes,
-    operation: FileTransferToken.Operation,
-    deadline: Optional[int] = None,
-    should_add_source: bool = False,
+    subtask_id: str = None,
+    source_package_path: str = None,
+    source_size: int = None,
+    source_package_hash: str = None,
+    result_package_path: str = None,
+    result_size: int = None,
+    result_package_hash: str = None,
+    authorized_client_public_key: bytes = None,
+    operation: FileTransferToken.Operation = None,
+    token_expiration_deadline: Optional[int] = None,
 ) -> FileTransferToken:
-    """
-    Function to create FileTransferToken from ReportComputedTask message
-    """
-    def calculate_token_expiration_deadline(
-        operation: FileTransferToken.Operation,
-        deadline: Optional[int],
-    ) -> int:
-        if deadline is not None:
-            token_expiration_deadline = get_current_utc_timestamp() + deadline
-        else:
-            if operation == FileTransferToken.Operation.upload:
-                token_expiration_deadline = (
-                        report_computed_task.task_to_compute.compute_task_def['deadline'] +
-                        3 * settings.CONCENT_MESSAGING_TIME +
-                        2 * calculate_maximum_download_time(report_computed_task.size)
-                )
-
-            elif operation == FileTransferToken.Operation.download:
-                token_expiration_deadline = (
-                        report_computed_task.task_to_compute.compute_task_def['deadline'] +
-                        calculate_subtask_verification_time(report_computed_task)
-                )
-        return token_expiration_deadline
-
-    task_id         = report_computed_task.task_to_compute.compute_task_def['task_id']
-    subtask_id      = report_computed_task.task_to_compute.compute_task_def['subtask_id']
-    file_path       = get_storage_result_file_path(task_id, subtask_id)
 
+    assert (source_size and source_package_hash) or (result_size and result_package_hash)
     assert isinstance(authorized_client_public_key, bytes)
-    assert isinstance(deadline, int) and not isinstance(deadline, bool) or deadline is None
-    assert isinstance(should_add_source, bool)
+    assert isinstance(token_expiration_deadline, int) and not isinstance(token_expiration_deadline, bool) or token_expiration_deadline is None
```

Comment:
`and not isinstance(token_expiration_deadline, bool)` this is not needed as you have already asserted that it is int.

Reply:
I remember now, I have reported it, keep it as it is.